### DISCUSSION TITLE
Troubleshooting low coverage report

### DISCUSSION THREAD

**Discussion by jozefizso** [vitest-dev/vitest#7145] [Q&A]
Hi, I have migrated the [generator-license](https://github.com/jozefizso/generator-license) library from CJS to ESM. As latest `yeoman-test` library uses vitest, I have switched from jest to vitest.

I am using the `@vitest/coverage-v8 2.1.8` for coverage.

Our coverage report went down to 33% from 100%.

Do you have any tips how to troubleshoot why some lines which are being executed during tests are not included in the coverage report?

Here is the PR with changes: https://github.com/jozefizso/generator-license/pull/129/files

Coveralls report: https://coveralls.io/builds/71518440

https://coveralls.io/builds/71518440/source?filename=app%2Findex.js#L19

**Comment by AriPerkkio** [comment]
How does `yeoman-test` run your code? Does it `import` it?

If it's using `require()` to load your code, Vite won't intercept it.

**Comment by jozefizso** [comment]
It appears the generator is imported during tests.

https://github.com/yeoman/environment/blob/main/src/store.ts#L57

As the `yeoman-test` is using sandboxing, it may interfere with code coverage reporting.
